### PR TITLE
Align enums to recent object changes

### PR DESCRIPTION
**Pull request by nguerrera**

### CODE REVIEW

File: adl/test/scenarios/v3/single/output/twitter/models/Poll.ts [hunk 1]
```diff
@@ -7,5 +7,5 @@ export interface Poll {
     end_datetime: dateTime;
     id?: string & RegularExpression<"^[0-9]{1,19}$">;
     options?: Array<PollOption> & MaximumElements<4> & MinimumElements<2>;
-    voting_status: unknown /*= (not tsschema -- undefined[object Object]/undefined ) =*/;
+    voting_status: "open" | "closed";
```
Comment (nguerrera) [inline adl/test/scenarios/v3/single/output/twitter/models/Poll.ts:10]: Example of desired change among all the renumbering :)

File: adl/core/model/schema/enum.ts [hunk 2]
```diff
@@ -90,73 +63,89 @@ class GenuineEnum extends TSSchema<EnumDeclaration> implements Enum {
...
+  constructor(node: EnumDeclaration, private anonymous: boolean) {
+    super('enum', node);
+    this.values = new CollectionImpl(this, this.addValue, this.removeValue, this.getValues);
+  }
 }
 
-export function createEnum(api: ApiModel, values: Array<EnumValue>, initializer?: Partial<Enum>) {
-  const name = initializer?.name ?? anonymous('enum');
+export function createEnum(api: ApiModel, identity: Identity, values: Array<EnumValue>, initializer?: Partial<Enum>) {
+  const { name, file } = api.getNameAndFile(identity, 'model');
```
Comment (fearthecowboy) [inline adl/core/model/schema/enum.ts:122]: should be 'enum'

Comment (fearthecowboy) [reply adl/core/model/schema/enum.ts:122]: this causes `model_311` in a test below.

Comment (fearthecowboy) [inline adl/core/model/schema/enum.ts:122]: test

File: adl/core/model/element.ts [hunk 3]
```diff
@@ -45,7 +45,6 @@ export class Initializer {
         proxy[key] = (<any>value);
       }
     }
```
Comment (fearthecowboy) [inline adl/core/model/element.ts:47]: uh, deleted a line?

File: adl/core/model/schema/enum.ts (hunk 2 above)
Comment (nguerrera) [reply adl/core/model/schema/enum.ts:122]: Sigh I managed to fix this without pushing it. I blame ready-for-lunch-brain :).

File: adl/core/model/element.ts (hunk 3 above)
Comment (nguerrera) [reply adl/core/model/element.ts:47]: The return this was causing a problem when I had a T1 | T2 and called initialize on it. But I don't have that anymore so reverting.

Comment (nguerrera) [reply adl/core/model/element.ts]: (We weren't using the return value AFAICT.)

Comment (nguerrera) [reply adl/core/model/element.ts]: Reverted

File: adl/core/model/schema/enum.ts (hunk 2 above)
Comment (nguerrera) [reply adl/core/model/schema/enum.ts]: Fixed

**Pull request merged**